### PR TITLE
Move "no SNI if IP address" responsibilty to TLS backends

### PR DESCRIPTION
**Pull request by pquentin**
* The ssl module already avoids SNI when the host is an IP address
 * pyOpenSSL now does that too
 * We were already using SNI on IP addresses with SecureTransport anyway

In other words, this does not change anything, but is cleaner
(we no longer test for SecureTransport in ssl.py) and will allow us to
lean on ssl.SSLContext to match hostnames.

### CODE REVIEW

File: src/urllib3/util/__init__.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
     PROTOCOL_TLS,
     SSLContext,
     assert_fingerprint,
+    is_ipaddress,
```
Comment (sethmlarson) [inline src/urllib3/util/__init__.py:14]: `urllib3.util` are public APIs, are we okay with having this function be public?

**Review by sethmlarson — APPROVED** [review]
The change itself looks good, great to simplify things! One thought:

File: src/urllib3/util/__init__.py (hunk 1 above)
Comment (pquentin) [reply src/urllib3/util/__init__.py]: Thanks, fixed! I hadn't realized that `util` was more public than `util.ssl_`. I certainly don't want to add more public APIs

**Comment by pquentin** [comment]
The codecov check is red, but the coverage is still 100%: https://app.codecov.io/gh/urllib3/urllib3/compare/2177/commits

**Comment by pquentin** [comment]
Was merged as part of #2178 by mistake, closing

**Pull request closed without merging**